jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java

```diff
@@ -47,7 +49,7 @@ protected void handleException(Throwable cause) {
     protected void handleException(Throwable cause,
                                    Map<String, Object> handlerInfoMap) {
         if (handlingProcessId != null && handlingStrategy != null) {
-            throw new ProcessWorkItemHandlerException(handlingProcessId, handlingStrategy, cause);
+            throw new ProcessWorkItemHandlerException(handlingProcessId, handlingStrategy, cause, retries);
```

Comment:
I see you added other constructors for `ProcessWorkItemHandlerException` as well, which have a default number of retries. This can cause issues in case number of retries is changed in retries field in AbstractLogOrThrowWorkItemHandler, but then the version of Exception constructor without retries is used. This will get out of sync, for example user then expects to set it for 5 retries, but then he forgets to pass it to the Exception constructor. Therefore, I think we should have retries parameter in all constructors of that exception; this will take it mandatory.

Reply:
OK, forget what I said here, then it would be confusing in case we use other handling strategies...